audio/filter/af_scaletempo2_internals.c

```diff
@@ -41,19 +41,15 @@ static bool in_interval(int n, struct interval q)
     return n >= q.lo && n <= q.hi;
 }
 
-static float **realloc_2d(float **p, int x, int y)
+static float **alloc_sample_buffer(struct mp_scaletempo2 *p, float **ptr, size_t size)
 {
-    float **array = realloc(p, sizeof(float*) * x + sizeof(float) * x * y);
-    float* data = (float*) (array + x);
-    for (int i = 0; i < x; ++i) {
-        array[i] = data + i * y;
-    }
-    return array;
-}
+    talloc_free(ptr);
```

Comment:
To keep this simple I decided to free the old buffer, since content is not needed. Is there a better way to do this?

Reply:
Reallocating this shouldn't be a problem. But you can also use `MP_TARRAY_GROW` here I guess, if you want to preserve allocation for smaller sizes. It will relink children.

Reply:
The problem is the size can change in both dimensions (`p->channels` or `size`), so it would need to track how many channels are already allocated and alloc/free the difference. That's why I thought I might as well free the whole thing, which is much simpler. It's also no performance problem because this is only done once for every buffer on filter init.

Reply:
I agree, it is doable to do, but in the same time it is not performance critical, so simple solution is fine.